### PR TITLE
fixes bug 1224011 - KeyError on histogram_date in Crashes per User

### PR DESCRIPTION
**Pull request by peterbe**
What do you think @AdrianGaudebert ?

### CODE REVIEW

**Comment by AdrianGaudebert** [comment]
If the error is caused by a parameter in the URL, don't you want to do a redirect instead? This way the user at least has a chance to see that something changed in his request. 

If you disagree, this is fine, r+.

**Comment by peterbe** [comment]
Actually, you have me convinced. I think I like that better. It's a little bit more transparent.

**Comment by peterbe** [comment]
This feels a lot better @AdrianGaudebert r?

File: webapp-django/crashstats/crashstats/tests/test_views.py [hunk 1]
```diff
@@ -2166,6 +2167,25 @@ def is_percentage(thing):
...
+        # If we use self.assertRedirects() it will actually go to the
+        # redirected URL which we haven't set up mocking for yet.
+        eq_(response.status_code, 302)
+        redirect_url = response['Location']
+        parsed = urlparse.urlparse(redirect_url)
+        eq_(parsed.path, url)
+        ok_('product=Whatever'in parsed.query)
+        ok_('foo=bar'in parsed.query)
+        ok_('date_range_type=build' not in parsed.query)
+        ok_('date_range_type=' not in parsed.query)
```
Comment (AdrianGaudebert) [inline webapp-django/crashstats/crashstats/tests/test_views.py]: These two lines are redundant, the first one is not needed.

**Comment by AdrianGaudebert** [comment]
r+

File: webapp-django/crashstats/crashstats/tests/test_views.py (hunk 1 above)
Comment (peterbe) [inline webapp-django/crashstats/crashstats/tests/test_views.py]: They are not redundant. They are the making sure that the new redirect URL doesn't have those things in there any more.

Comment (AdrianGaudebert) [inline webapp-django/crashstats/crashstats/tests/test_views.py]: Just for the sake of arguing: if ``date_range_type=`` is not in ``parsed.query``, ``date_range_type=build`` definitely is not in it either. :) So the second line covers both cases, and the first one is unneeded.

Comment (AdrianGaudebert) [inline webapp-django/crashstats/crashstats/tests/test_views.py]: (Though please do not bother changing that, it's really not worth our time. )

Comment (peterbe) [inline webapp-django/crashstats/crashstats/tests/test_views.py]: NOW I see what you mean. Thank you for not giving up on me. We can just keep the ` ok_('date_range_type=' not in parsed.query)` line.

Comment (peterbe) [inline webapp-django/crashstats/crashstats/tests/test_views.py]: oh. I merged it already :)

**Pull request merged**